queue/queue.go

```diff
@@ -13,81 +15,56 @@ import (
 	"github.com/streadway/amqp"
 )
 
-func (qm *Manager) setupLogging() error {
-	logFileName := fmt.Sprintf("/var/log/temporal/%s_serice.log", qm.QueueName)
+func (qm *Manager) setupLogging(logFilePath ...string) error {
+	var logFileName string
+	if len(logFilePath) > 0 {
+		logFileName = fmt.Sprintf("%s-%s_service.log", logFilePath[0], qm.QueueName)
+	} else {
+		logFileName = fmt.Sprintf("/var/log/temporal/%s_service.log", qm.QueueName)
+	}
 	logFile, err := os.OpenFile(logFileName, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0640)
 	if err != nil {
 		return err
 	}
 	logger := log.New()
 	logger.Out = logFile
-	qm.Logger = logger
-	qm.Logger.Info("Logging initialized")
-	return nil
-}
-
-func (qm *Manager) parseQueueName(queueName string) error {
-	host, err := os.Hostname()
-	if err != nil {
-		return err
-	}
-	qm.QueueName = fmt.Sprintf("%s+%s", host, queueName)
+	qm.logger = logger
+	qm.logger.Info("Logging initialized")
 	return nil
 }
 
 // Initialize is used to connect to the given queue, for publishing or consuming purposes
-func Initialize(queueName, connectionURL string, publish, service bool) (*Manager, error) {
+func Initialize(queueName, connectionURL string, publish, service bool, logFilePath ...string) (*Manager, error) {
```

Comment:
```suggestion
func New(queue, URL string, publish, service bool, logFilePath ...string) (*Manager, error) {
```

also, why is `service` a `bool`? it looks like it enables logging